apps/rush-lib/src/cli/actions/BaseRushAction.ts

```diff
@@ -12,26 +12,42 @@ import { LockFile } from '@microsoft/node-core-library';
 
 import RushConfiguration from '../../data/RushConfiguration';
 import EventHooksManager from '../logic/EventHooksManager';
+import RushCommandLineParser from './RushCommandLineParser';
 
 export interface IRushCommandLineActionOptions extends ICommandLineActionOptions {
   /**
    * If true, no locking mechanism will be enforced when this action is run.
    * Note this defaults to false (which is a safer assumption in case this value
-   *  is ommitted).
+   *  is omitted).
    */
   safeForSimultaneousRushProcesses?: boolean;
+
+  /**
+   * The rush parser.
+   */
+  parser: RushCommandLineParser;
 }
 
 /**
  * The base Rush action that all Rush actions should extend.
  */
 export abstract class BaseRushAction extends CommandLineAction {
-  private _rushConfiguration: RushConfiguration;
+  private _parser: RushCommandLineParser;
   private _eventHooksManager: EventHooksManager;
   private _safeForSimultaneousRushProcesses: boolean;
```

Comment:
>_safeForSimultaneousRushProcesses [](start = 10, length = 33)

_parser and _safeForSimultaneousRushProcesses could be readonly

Reply:
They effectively are since they're only accessible through their getters.

---
In reply to: [178236819](https://github.com/Microsoft/web-build-tools/pull/585#discussion_r178236819) [](ancestors = 178236819)